source/server/microkernel_architecture/microkernel_modules/database_manager/Entity.hpp

```diff
@@ -0,0 +1,103 @@
+#ifndef ENTITY_HPP_
+# define ENTITY_HPP_
+
+# include <QSqlDatabase>
+# include <QObject>
+# include <QMetaObject>
+# include <QMetaProperty>
+# include <QStringList>
+# include <QVariant>
+# include <QString>
+# include <QVector>
+# include <QSqlRecord>
+# include <QSqlQuery>
+# include "QueryBuilderMySql.hpp"
+
+# define MYSQL_TYPE "MYSQL"
+
+class Entity : public QObject
+{
+    Q_OBJECT
+
+private:
+    QSqlDatabase *_db;
+    QString _dbType;
+    QVector<QString> *_propertiesName;
+    QVector<QString> *_propertiesValue;
+    bool connect();
+    bool startConnection();
+    void getAllProperties();
+    QSqlQuery *prepareInsert() const;
+    QSqlQuery *prepareUpdate() const;
+
+protected:
+    QString _table;
+    const QString &getTable() const;
+    void setTable(const QString &newTable);
+
+public:
+    Entity();
+    virtual ~Entity();
+    bool save();
+    bool deleteQuery(AQueryBuilder *builder);
+    AQueryBuilder *getQueryBuilder() const;
+
+
+    //
+    // Find and return the entity by its id
+    // Function in .hpp because of templating
+    //
+    template<class T>
+    T *find(int id)
+    {
+      if (!startConnection())
+	return NULL;
+
+      AQueryBuilder *builder = getQueryBuilder();
+      builder->select("*")
+	->where("id = :id")
+	->bindValue(":id", id)
+	;
+
+      QVector<T *> result = this->request<T>(builder);
+
+      if (result.size() == 0)
+	return NULL;
+
+      return result.at(0);
+    }
+
+    //
+    // Send a query request and return the result
+    // Function in .hpp because of templating
+    //
+    template<class T>
+    QVector<T *> request(AQueryBuilder *builder)
+	{
+	    QVector<T *> result;
+	    QSqlRecord recQuery;
+	    QSqlQuery *query;
+
+	    if (!startConnection())
+		return (result);
+	    query = builder->build();
+	    query->exec();
+	    while (query->next())
+	    {
+		recQuery = query->record();
+		T *entity = new T();
+		for (auto prop : _propertiesName->toStdVector())
+		{
+		    int index = recQuery.indexOf(prop);
+		    if (index < 0)
+			continue;
+		    entity->setProperty(prop.toLocal8Bit().constData(),
+					query->value(index));
+		}
+		result.push_back(entity);
+	    }
+	    return (result);
+	}
```

Comment:
The indentation here has the good format. A strange format, yes, but it's because this function is coded IN the class declaration.